File: SharePaint/src/app/white-board/white-board/Models/stroke.ts
export class Stroke {
    Id:string;
    StartPoint: Point;
    EndPoint : Point;
    Lines: Line[];
    LinesQueue: Line[];
    
    IsUndone: boolean;
    constructor()
    {
        this.StartPoint = new Point();
        this.EndPoint = new Point();
        this.Lines = [];
        this.LinesQueue = [];
    }
}

export class Point {
    X:number;
    Y:number;
}

export class Line {
    StrokeId:string;
    StartPoint: Point;
    EndPoint : Point;
    constructor()
    {
        this.StartPoint = new Point();
        this.EndPoint = new Point();
    }
}

export class Paint{
      Id:string;
      Name : string;
     Strokes: Stroke[];
}

File: SharePaint.Services/SharePaint.Data/Models/Stroke.cs
using System;
using System.Collections.Generic;

namespace SharePaint.Data.Models
{
    public class Stroke
    {
        public Guid Id { get; set; }
        public Guid PaintId { get; set; }
        public Guid UserId { get; set; }
        public Point StartPoint { get; set; }
        public Point EndPoint { get; set; }
        public bool IsUndone{get;set;}
        public IEnumerable<Line> Lines { get; set; }
    }
}
